app/src/main/java/com/pajato/android/gamechat/exp/TileClickHandler.java

```diff
@@ -74,8 +94,22 @@ private boolean isPlayingOutOfTurn(final int position) {
         Team team = board.getTeam(position);
         if (team == Team.NONE)
             return false;
+
+        // Ensure that the current user is choosing a piece that belongs to their team. If they are
+        // playing with a "Friend", then we don't need to worry about confirming players' identities
+        Team playerTeam = team;
+        for (Player player :  mModel.getPlayers()) {
+            if(player.id == null && player.name.equals("Friend")) {
```

Comment:
It would be better to use the play mode in the condition rather than the player's name.